app/controllers/schemes_controller.rb

```diff
@@ -48,7 +48,10 @@ def update
 
     validation_errors scheme_params
     if @scheme.errors.empty? && @scheme.update(scheme_params)
-      if check_answers
+      if scheme_params[:confirmed] == "true"
+        @scheme.locations.each {|location| location.update!(confirmed:true)}
```

Comment:
It's possible to shorten this slightly to:

```suggestion
        @scheme.locations.update!(confirmed: true)
```